tempodb/wal/wal.go

```diff
@@ -135,7 +135,14 @@ func (w *WAL) NewFile(blockid uuid.UUID, tenantid string, dir string, name strin
 	if err != nil {
 		return nil, err
 	}
-	return os.OpenFile(filepath.Join(p, fmt.Sprintf("%v:%v:%v", blockid, tenantid, name)), os.O_CREATE|os.O_RDWR, 0644)
+
+	// blockID, tenantID, version, encoding (compression), dataEncoding, searchFileSuffix
+	filename := fmt.Sprintf("%v:%v:%v:%v:%v:%v", blockid, tenantid, "v2", backend.EncNone, "", name)
```

Comment:
It might be better to move the filename parsing code from rescan_blocks.go to here, and in the future this can cover the file name generation and parsing for the regular wal files too.

Reply:
I agree. At the moment, the regular wal files already have a `parseFilename()` in [append_block.go](https://github.com/grafana/tempo/blob/6e94c1ffc10c4c817d53f0feb5c23afad11063aa/tempodb/wal/append_block.go#L263) that has some logic around checking the number of splits etc, for backward compatibility reasons when we added new versions. We could simplify that function now that we have dropped support for those older versions.

Reply:
btw, thoughts on dropping the "searchdata" suffix as the sixth split in the filename? we have a separate directory for search data so we shouldn't need that anymore

Reply:
comment addressed. also, dropped the "searchdata" suffix.